### PR TITLE
Nu 957 error handling

### PR DESCRIPTION
**Pull request by Dzuming**
- Provide "No network" error
- Provide "No backend connection" error
- Provide e2e tests for the above logic
- Some small code cleaning

NOTES:
- We discussed with a team that  No backend is available because of update case is something non-trivial and still for discussion, that's why it's moved to the separate task
- The easiest way to test the first case is to disable your network connection

image

### CODE REVIEW

File: designer/client/src/containers/ConnectionErrorProvider/ConnectionErrorProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    };
+
+    const connectionErrorComponent = useMemo(() => {
+        switch (connectionError) {
+            case "NO_NETWORK_ACCESS": {
+                return (
+                    <ConnectionErrorContent
+                        headerText={"No network access"}
+                        contentText={
+                            "We're sorry, but it appears that you don't have a network connection at the moment. Please check your network settings."
```
Comment (lciolecki) [inline designer/client/src/containers/ConnectionErrorProvider/ConnectionErrorProvider.tsx:24]: Missing transialtions my friend!

Comment (Dzuming) [reply designer/client/src/containers/ConnectionErrorProvider/ConnectionErrorProvider.tsx]: Done! ✔️

File: designer/client/src/containers/ConnectionErrorProvider/ConnectionErrorProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            {children}
+        </ConnectionErrorContext.Provider>
+    );
+};
+
+export const useChangeConnectionError = () => {
+    const context = useContext(ConnectionErrorContext);
+
+    if (!context) {
+        throw new Error(`${useChangeConnectionError.name} was used outside of its ${ConnectionErrorContext.displayName} providerł`);
```
[the remark this answers is not in the document]
Comment (Dzuming) [reply designer/client/src/containers/ConnectionErrorProvider/ConnectionErrorProvider.tsx]: Done! ✔️

File: designer/client/src/http/HttpService.ts [hunk 3]
```diff
@@ -123,12 +121,23 @@ class HttpService {
...
             .get<BackendNotification[]>("/notifications")
-            .then((d) => d.data)
+            .then((d) => {
+                onSuccess();
+                return d.data;
+            })
             .catch((error) => {
-                this.#addError(i18next.t("notification.error.cannotFetchBackendNotifications", "Cannot fetch backend notification"), error);
+                onError(error, () => {
+                    this.#addError(
```
Comment (lciolecki) [inline designer/client/src/http/HttpService.ts:136]: Hmm, do we need this `#` before method name?

**Comment by lciolecki** [comment]
Nice message presentation, I like this!

**Review by lciolecki — APPROVED** [review]

File: designer/client/src/http/HttpService.ts (hunk 3 above)
Comment (Dzuming) [reply designer/client/src/http/HttpService.ts:136]: I didn't create this error, but I assume, someone wants to make this variable private https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields so i would say, yes :)

**Comment by lciolecki** [comment]
I wonder about the service worker and offline web page like here: https://medium.com/bliblidotcom-techblog/offline-web-page-access-with-service-worker-f973acd25648, WDYT?

File: designer/client/src/http/HttpService.ts (hunk 3 above)
Comment (lciolecki) [reply designer/client/src/http/HttpService.ts:136]: Ehh java script.. :) Let's leave it.

**Comment by Dzuming** [comment]
> I wonder about the service worker and offline web page like here: https://medium.com/bliblidotcom-techblog/offline-web-page-access-with-service-worker-f973acd25648, WDYT?

Yeah, we even talked about it with @JulianWielga and for sure, it's an option to consider, but for a different mr for sure ;)

**Review by JulianWielga — APPROVED** [review]

**Pull request merged**